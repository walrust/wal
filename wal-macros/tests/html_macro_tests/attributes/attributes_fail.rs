use wal_macros::html;

include!("../utils/non_display_struct.rs");

fn main() {
    html! { <div attr></div> };
    html! { <div attr=></div> };
    html! { <div attr "value"></div> };
    html! { <div attr="key1" attr="key2"></div> };
    html! { <div attr1= attr2="key2"></div> };
    html! { <div attr={NonDisplayStruct}></div> };
}
