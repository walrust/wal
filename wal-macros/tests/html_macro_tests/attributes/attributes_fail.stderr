error: expected `=`
 --> tests/html_macro_tests/attributes/attributes_fail.rs:4:22
  |
4 |     html! { <div attr></> };
  |                      ^

error: Expected a literal or an expression block
 --> tests/html_macro_tests/attributes/attributes_fail.rs:5:23
  |
5 |     html! { <div attr=></> };
  |                       ^

error: expected `=`
 --> tests/html_macro_tests/attributes/attributes_fail.rs:6:23
  |
6 |     html! { <div attr "value"></> };
  |                       ^^^^^^^

error: Duplicate attribute `attr`
 --> tests/html_macro_tests/attributes/attributes_fail.rs:7:30
  |
7 |     html! { <div attr="key1" attr="key2"></> };
  |                              ^^^^

error: Expected a literal or an expression block
 --> tests/html_macro_tests/attributes/attributes_fail.rs:8:25
  |
8 |     html! { <div attr1= attr2="key2"></> };
  |                         ^^^^^
